core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -169,6 +172,32 @@ class CloudSqlInstance {
     }
   }
 
+  private OAuth2Credentials parseCredentials(HttpRequestInitializer source) {
+    if (source instanceof HttpCredentialsAdapter) {
+      HttpCredentialsAdapter adapter = (HttpCredentialsAdapter) source;
+      return (OAuth2Credentials) adapter.getCredentials();
+    }
+
+    if (source instanceof Credential) {
+      Credential credential = (Credential) source;
+      AccessToken accessToken = new AccessToken(
+          credential.getAccessToken(),
+          new Date(credential.getExpirationTimeMilliseconds())
+      );
+      GoogleCredentials googleCredentials = new GoogleCredentials(accessToken) {
+        @Override
+        public AccessToken refreshAccessToken() throws IOException {
+          credential.refreshToken();
+          return new AccessToken(credential.getAccessToken(), new Date(credential.getExpirationTimeMilliseconds()));
```

Comment:
Looks like the linter is failing on this line:
```suggestion
          return new AccessToken(
               credential.getAccessToken(),
               new Date(credential.getExpirationTimeMilliseconds())
          );
```

Reply:
Will fix it